src/main/scala/ai/privado/languageEngine/javascript/tagger/collection/CollectionTagger.scala

```diff
@@ -91,17 +95,21 @@ class CollectionTagger(cpg: Cpg, ruleCache: RuleCache) extends PrivadoParallelCp
     val objectKeyValuePairs      = block.astChildren.isCall.name("<operator>.assignment").l
     var handlerMethod: MethodRef = null
     var path: String             = ""
-    for (keyVal <- objectKeyValuePairs) {
-      if (keyVal.astChildren.nonEmpty) {
-        val objKeyVal = keyVal.astChildren
-        val key       = objKeyVal.isCall.head
+    if (objectKeyValuePairs.nonEmpty && objectKeyValuePairs.isInstanceOf[List[Call]]) {
+      for (keyVal <- objectKeyValuePairs) {
+        if (keyVal.astChildren.nonEmpty && keyVal.astChildren.isInstanceOf[Traversal[AstNode]]) {
+          val objKeyVal = keyVal.astChildren
+          val key       = objKeyVal.isCall
 
-        if (key.code.contains(pathField) && key.astSiblings.head.isLiteral) {
-          path = key.astSiblings.head.code
-        }
+          if (key.nonEmpty) {
```

Comment:
is the code inside is getting executed as `key` is a traversal/iterator, which can only be used once?